sentry-core/src/main/java/io/sentry/core/EnvelopeSender.java

```diff
@@ -111,7 +111,7 @@ private void processEnvelope(SentryEnvelope envelope, CachedEnvelopeHint hint)
               continue;
             }
             hub.captureEvent(event, hint);
-            hint.waitFlush();
+            hint.waitFlush(); // why do we need to waitFlush? is not ok to be async?
```

Comment:
@bruno-garcia why don't we just enqueue the events and the rest of the flow keeps the same?

Reply:
We need to know if it succeeded or not to delete the file.
If it fails due to the device being offline we don't want to delete it.
Since this is a background thread anyway, it's fine to sync with the transport